### PR TITLE
TeslaGate API and Event changed

### PR DESCRIPTION
**Pull request by louis1706**
- Make easyer for patch it
- Repair the actual idle bug
- Add some method in TeslaGate API

### CODE REVIEW

**Comment by louis1706** [comment]
He have a actual bug please don't merged it

File: Exiled.Events/EventArgs/TriggeringTeslaEventArgs.cs [hunk 1]
```diff
@@ -21,14 +21,12 @@ public class TriggeringTeslaEventArgs : EventArgs
         /// </summary>
         /// <param name="player"><inheritdoc cref="Player"/></param>
         /// <param name="teslaGate"><inheritdoc cref="Tesla"/></param>
-        /// <param name="isInHurtingRange"><inheritdoc cref="IsInHurtingRange"/></param>
-        /// <param name="isTriggerable"><inheritdoc cref="IsTriggerable"/></param>
-        public TriggeringTeslaEventArgs(Player player, TeslaGate teslaGate, bool isInHurtingRange, bool isTriggerable)
```
Comment (joker-119) [inline Exiled.Events/EventArgs/TriggeringTeslaEventArgs.cs:26]: Why would we remove these parameters, just to re-run the same methods for them? It's inefficienct, especially considering each of these methods preforms a Math.sqrt operation.

Comment (louis1706) [reply Exiled.Events/EventArgs/TriggeringTeslaEventArgs.cs:26]: it's not for re-run i have thinking of only run it in the event and don't run this method in the normal method

Comment (joker-119) [reply Exiled.Events/EventArgs/TriggeringTeslaEventArgs.cs:26]: Oh I misread the patch part, my bad

Comment (louis1706) [reply Exiled.Events/EventArgs/TriggeringTeslaEventArgs.cs:26]: yes it's the reason why i have made the prefix just for explain how it's could be ^^'

**Comment by louis1706** [comment]
Nao wante change:
- re-added the Method ``CanBeTriggered``
- Trigger method not going to check if it's already in progress
Method name
- Changing for original name of the method

**Comment by louis1706** [comment]
In Favor of Nao PR

**Pull request merged**